test/UtteranceTest.ts

```diff
@@ -203,5 +207,27 @@ describe("UtteranceTest", function() {
             assert.isTrue(utterance.matched());
             assert.equal(utterance.intent(), "NumberSlot");
         });
+
+        it("Matches with symbols in the phrase", () => {
+            const utterance = new Utterance(model, "good? #%.morning");
```

Comment:
The tests should also include an utterance with:
,
-
:
Those are very common punctuation marks

Reply:
added one more test for this one, were missing the character exclusions too